geoh5py/objects/surveys/electromagnetics/base.py

```diff
@@ -193,10 +194,70 @@ def components(self) -> dict | None:
 
         return None
 
+    def copy(
+        self,
+        parent=None,
+        copy_children: bool = True,
+        clear_cache: bool = False,
+        mask: np.ndarray | None = None,
+        cell_mask: np.ndarray | None = None,
+        apply_to_complement: bool = True,
+        **kwargs,
+    ):
+        """
+        Sub-class extension of :func:`~geoh5py.objects.cell_object.CellObject.copy`.
+        """
+        if parent is None:
+            parent = self.parent
+
+        omit_list = [
+            "_metadata",
+            "_receivers",
+            "_transmitters",
+            "_base_stations",
+            "_tx_id_property",
+        ]
+        kwargs["omit_list"] = omit_list
+        metadata = self.metadata.copy()
+        new_entity = super().copy(
+            parent=parent,
+            clear_cache=clear_cache,
+            copy_children=copy_children,
+            mask=mask,
+            cell_mask=cell_mask,
+            **kwargs,
+        )
+
+        metadata["EM Dataset"][new_entity.type] = new_entity.uid
+
+        if apply_to_complement and (getattr(self, "complement", None) is not None):
+            base_object = (
+                self.base_transmitter_type
+                if isinstance(self, self.default_receiver_type)
+                else self.base_receiver_type
+            )
+            new_complement = super(base_object, self.complement).copy(  # type: ignore
```

Comment:
Damn, that's nice.

Reply:
Maybe `complement` should be default to None on the base class?

Reply:
Done